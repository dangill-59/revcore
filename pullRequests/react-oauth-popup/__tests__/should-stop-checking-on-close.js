import OauthPopup from '../src/index.js';
import React from 'react';
import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('OauthPopup stops checking for code when popup closes', () => {
  const mockCallBack = jest.fn();
  global.open = jest.fn().mockReturnValue({
    location: 'http://www.google.com?code=amazing',
  });

  const component = renderer.create(
    <OauthPopup onCode={mockCallBack}>
      <div>coooooooool</div>
      <h2>siiiiiiiick</h2>
      <h2>leeegiiiiit</h2>
      <button>Doooooope</button>
    </OauthPopup>,
  );
  let popup = component.toJSON();
  expect(popup).toMatchSnapshot();
  popup.props.onClick();
  popup = component.toJSON();
  expect(popup).toMatchSnapshot();
  expect(global.open).toBeCalled();
  jest.runAllTimers();
  expect(setInterval).toHaveBeenCalledTimes(1);
  expect(clearInterval).toHaveBeenCalledTimes(1);
  expect(mockCallBack).toHaveBeenCalledTimes(1);
});
