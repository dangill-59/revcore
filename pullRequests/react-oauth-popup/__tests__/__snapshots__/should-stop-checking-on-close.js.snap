// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OauthPopup stops checking for code when popup closes 1`] = `
<div
  onClick={[Function]}
>
   
  <div>
    coooooooool
  </div>
  <h2>
    siiiiiiiick
  </h2>
  <h2>
    leeegiiiiit
  </h2>
  <button>
    Doooooope
  </button>
   
</div>
`;

exports[`OauthPopup stops checking for code when popup closes 2`] = `
<div
  onClick={[Function]}
>
   
  <div>
    coooooooool
  </div>
  <h2>
    siiiiiiiick
  </h2>
  <h2>
    leeegiiiiit
  </h2>
  <button>
    Doooooope
  </button>
   
</div>
`;
